src/abi.h

```diff
@@ -211,7 +211,7 @@ struct netplan_net_definition {
     /* status options */
     gboolean optional;
     NetplanOptionalAddressFlag optional_addresses;
-    gboolean critical;
+    char* keep_configuration;
```

Comment:
To make things simpler, we should use an `enum` for this option. It has a well defined set of possible values, so it would naturally fit well in an `enum`:

```c
enum {
    NETPLAN_CRITICAL_TRUE,
    NETPLAN_CRITICAL_FALSE,
    NETPLAN_CRITICAL_STATIC,
    ...,
} CriticalOptions;
```

Please, check my comment in the summary.

Reply:
Good idea! Let me revamp

Reply:
I've been trying to rework this, but I'm not sure if I'm on the right track, as quite a few tests are failing or even segfaulting.